java/ql/src/Telemetry/DatabaseQualityDiagnostics.ql

```diff
@@ -0,0 +1,40 @@
+/**
+ * @name Low Java analysis quality
+ * @description Low Java analysis quality
+ * @kind diagnostic
+ * @id java/diagnostic/database-quality
+ */
+
+import java
+import DatabaseQuality
+
+private newtype TDbQualityDiagnostic =
+  TTheDbQualityDiagnostic() {
+    exists(float percentageGood |
+      CallTargetStatsReport::percentageOfOk(_, percentageGood)
+      or
+      ExprTypeStatsReport::percentageOfOk(_, percentageGood)
+    |
+      percentageGood < 95
+    )
+  }
+
+class DbQualityDiagnostic extends TDbQualityDiagnostic {
+  string toString() {
+    result =
+      "There were significant issues scanning Java code. " +
+        "This may be caused by problems identifying dependencies or use of generated source code, among other reasons -- "
+        + "see other CodeQL diagnostics reported here for more details of possible causes. " +
+        "This may lead to false-positive or missing results. Consider analysing Java using the `autobuild` or `manual` build modes."
```

Comment:
It might be worth checking whether the Markdown formatting here renders properly in code scanning — since we don't support Markdown for alert messages, I have a suspicion it might end up being rendered as plain text.

Reply:
It does work, e.g. https://github.com/smowton/test-diagnostics-from-ql/security/code-scanning/tools/CodeQL/status/configurations/api/0a0e717e7702ad6128802b437a72832333465cec67163ebe61b8292092f457c4

Reply:
Excellent, thanks for checking!